lib/rack/lint.rb

```diff
@@ -462,15 +462,33 @@ def check_environment(env)
         ##
         ## ==== <tt>rack.response_finished</tt>
         ##
-        ## If present, an array of callables that will be run by the server after the response has been processed. The callables are called with <tt>environment, status, headers, error</tt> arguments and should not raise any exceptions. The callables would typically be called after sending the response to the client, but it could also be called if an error occurs while generating the response or sending the response (in that case, the +error+ argument will be a kind of +Exception+). The callables will be called in reverse order.
+        ## If present, an array of callables that will be run by the server after the response has been processed. \
         if rack_response_finished = env[RACK_RESPONSE_FINISHED]
           raise LintError, "rack.response_finished must be an array of callable objects" unless rack_response_finished.is_a?(Array)
-          rack_response_finished.each do |callable|
+
+          rack_response_finished.map! do |callable|
             raise LintError, "rack.response_finished values must respond to call(env, status, headers, error)" unless callable.respond_to?(:call)
+
+            ResponseFinishedWrapper.new(callable)
           end
         end
       end
 
+      class ResponseFinishedWrapper
+        def initialize(callback)
+          @callback = callback
+        end
+
+        ## The callables are called with <tt>environment, status, headers, error</tt> arguments and should not raise any exceptions. The callables would typically be called after sending the response to the client, but it could also be called if an error occurs while generating the response or sending the response (in that case, the +error+ argument will be a kind of +Exception+). The callables will be called in reverse order.
+        def call(*args)
+          unless args.length == 4
```

Comment:
I agree with adding this check, but it may be better to check the types of each argument instead of just checking the arity.

Reply:
I think that's a good idea as well, I have to run for now but I'll work on it